src/MRA.Jobs.Api/Application/Features/TrainingVacancies/Queries/GetTrainingCategoriesQueryHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MRA.Jobs.Application.Contracts.TrainingVacancies.Queries;
+using MRA.Jobs.Application.Contracts.TrainingVacancies.Responses;
+using MRA.Jobs.Application.Contracts.VacancyCategories.Responses;
+
+namespace MRA.Jobs.Application.Features.TrainingVacancies.Queries;
+public class GetTrainingCategoriesQueryHandler : IRequestHandler<GetTrainingCategoriesQuery, List<TrainingCategoriesResponce>>
+{
+    IApplicationDbContext _context;
+    IMapper _mapper;
+    public GetTrainingCategoriesQueryHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+    public async Task<List<TrainingCategoriesResponce>> Handle(GetTrainingCategoriesQuery request, CancellationToken cancellationToken)
+    {
+        var trainings = await _context.TrainingVacancies.ToListAsync();
+
+        if (request.CheckDate)
+        {
+            DateTime now = DateTime.UtcNow;
+            trainings = (from t in trainings
+                         where t.PublishDate <= now && t.EndDate >= now
+                         select t).ToList();
+        }

```

Comment:
this is not performant. You should not do always .ToList. Also, just use simple Linq query

Reply:
The type of trainings variable is list. How can i do it without using Tolist()?